cfn-resources/datalakes/cmd/resource/resource.go

```diff
@@ -440,26 +420,26 @@ func dataLakeIsReady(client *mongodbatlas.Client, projectID, name, targetState s
 		}
 		return false, "", err
 	}
-	return dataLake.State == targetState, dataLake.State, nil
+	return *dataLake.State == targetState, *dataLake.State, nil
```

Comment:
can dataLake be nil so *dataLake will panic?

Reply:
I am not sure. I updated the logic to check for nill